packages/core/database/src/schema/builder.ts

```diff
@@ -273,6 +264,29 @@ const createHelpers = (db: Database) => {
         dropColumn(tableBuilder, removedColumn);
       }
 
+      // for mysql only, dropForeignKey also removes the index, so don't drop it twice
+      const isMySQL = db.config.connection.client === 'mysql';
+      const ignoreForeignKeyNames = isMySQL
+        ? [
+            ...table.foreignKeys.removed.map((fk) => fk.name),
```

Comment:
Are we sure the fk and idx have the same names 🤔  don't we do _fk or _idx or sth like that ?

Reply:
I might have something wrong here but from what I can tell:

There's a foreign key index, and a foreign key constraint. When we create the foreign key, we just give it the index name and they both use it.

[When we create them, we use the same name for both](https://github.com/strapi/strapi/blob/922ac347bb655c5d2cfd5f70ecf727f8199372fe/packages/core/database/src/schema/schema.ts#L94).

For postgres and sqlite, `dropForeignKey` drops the constraint, but not the index. For mysql, `dropForeignKey` drops both.

Reply:
👍